src/internal/m365/collection/groups/collection.go

```diff
@@ -248,3 +249,211 @@ func (col *prefetchCollection[C, I]) streamItems(ctx context.Context, errs *faul
+	}
+
+	semaphoreCh := make(chan struct{}, col.Opts().Parallelism.ItemFetch)
+	defer close(semaphoreCh)
+
+	// delete all removed items
+	for id := range col.removed {
+		semaphoreCh <- struct{}{}
+
+		wg.Add(1)
+
+		go func(id string) {
+			defer wg.Done()
+			defer func() { <-semaphoreCh }()
+
+			col.stream <- data.NewDeletedItem(id)
+
+			atomic.AddInt64(&streamedItems, 1)
+			col.Counter.Inc(count.StreamItemsRemoved)
+
+			if colProgress != nil {
+				colProgress <- struct{}{}
+			}
+		}(id)
+	}
+
+	// add any new items
+	for id, modTime := range col.added {
+		if el.Failure() != nil {
+			break
+		}
+
+		wg.Add(1)
+		semaphoreCh <- struct{}{}
+
+		go func(id string, modTime time.Time) {
+			defer wg.Done()
+			defer func() { <-semaphoreCh }()
+
+			ictx := clues.Add(
+				ctx,
+				"item_id", id,
+				"parent_path", path.LoggableDir(col.LocationPath().String()))
+
+			col.stream <- data.NewLazyItemWithInfo(
+				ictx,
+				&lazyItemGetter[C, I]{
+					modTime:       modTime,
+					getAndAugment: col.getAndAugment,
+					userID:        col.protectedResource,
+					itemID:        id,
+					containerIDs:  col.FullPath().Folders(),
+					contains:      col.contains,
+					parentPath:    col.LocationPath().String(),
+				},
+				id,
+				modTime,
+				col.Counter,
+				el)
+
+			// Need remaining things, copy from prefetch collection
+			// totalBytes - not available since info gets called later now
+			atomic.AddInt64(&streamedItems, 1)
+
+			if colProgress != nil {
+				colProgress <- struct{}{}
+			}
+		}(id, modTime)
+	}
+
+	wg.Wait()
+}
+
+type lazyItemGetter[C graph.GetIDer, I groupsItemer] struct {
+	getAndAugment getItemAndAugmentInfoer[C, I]
+	userID        string
+	itemID        string
+	parentPath    string
+	containerIDs  path.Elements
+	modTime       time.Time
+	contains      container[C]
+}
+
+func (lig *lazyItemGetter[C, I]) GetData(
+	ctx context.Context,
+	errs *fault.Bus,
+) (io.ReadCloser, *details.ItemInfo, bool, error) {
+	writer := kjson.NewJsonSerializationWriter()
+	defer writer.Close()
+
+	item, info, err := lig.getAndAugment.getItem(
+		ctx,
+		lig.userID,
+		lig.containerIDs,
+		lig.itemID)
+	if err != nil {
+		// If an item was deleted then return an empty file so we don't fail
+		// the backup and return a sentinel error when asked for ItemInfo so
+		// we don't display the item in the backup.
+		//
+		// The item will be deleted from kopia on the next backup when the
+		// delta token shows it's removed.
+		if graph.IsErrDeletedInFlight(err) {
+			logger.CtxErr(ctx, err).Info("item not found")
+			return nil, nil, true, nil
+		}
+
+		err = clues.Wrap(err, "getting item data").Label(fault.LabelForceNoBackupCreation)
+		errs.AddRecoverable(ctx, err)
+
+		return nil, nil, false, err
+	}
+
+	lig.getAndAugment.augmentItemInfo(info, lig.contains.container)
+
+	if err := writer.WriteObjectValue("", item); err != nil {
+		err = clues.Wrap(err, "writing item to serializer").Label(fault.LabelForceNoBackupCreation)
```

Comment:
WC needed here for telemetry, since we don't hand ctx to the called func.
```suggestion
		err = clues.WrapWC(ctx, err, "writing item to serializer").Label(fault.LabelForceNoBackupCreation)
```

Some other error handling below also needs this.